core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/avro/AvroInlet.scala

```diff
@@ -35,3 +36,18 @@ object AvroInlet {
   def create[T <: SpecificRecordBase](name: String, clazz: Class[T]): AvroInlet[T] =
     AvroInlet[T](name)(ClassTag.apply(clazz))
 }
+
+// Additional AvroInlet reading from all partitions of the topic. This is done by ensuring unique
+// groupID for Kafka listener
+case class AvroInletAllPartitions[T <: SpecificRecordBase: ClassTag](name: String) extends CodecInlet[T] {
```

Comment:
Instead it would be more consistent to add a `withUniqueGroupId` to `CodecInlet`. We already have a 'builder' API for `CodecOutlet`, which allows for setting a partitioner with `withPartitioner`. So doing it that way would be more consistent. Also, that way, this class is not necessary anymore.

Reply:
Redesign based on Ray's suggestion of implementing a builder